### PR TITLE
Use zval_get_tmp_string method

### PR DESCRIPTION
**Pull request by JakubOnderka**

### CODE REVIEW

File: redis_commands.c [hunk 1]
```diff
@@ -1780,44 +1780,33 @@ int redis_key_varval_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
...
-    redis_cmd_append_sstr_key(&cmdstr, ZSTR_VAL(zstr), ZSTR_LEN(zstr), redis_sock, slot);
-    zend_string_release(zstr);
+    redis_cmd_append_sstr_key_zstr(&cmdstr, key, redis_sock, slot);
 
     /* Add members */
-    for (i = 1; i < argc; i++ ){
-        redis_cmd_append_sstr_zval(&cmdstr, &z_args[i], redis_sock);
+    for (i = 0; i < argc; i++) {
+        redis_cmd_append_sstr_zval(&cmdstr, args, redis_sock);
+        args++;
```
Comment (yatsukhnenko) [inline redis_commands.c]:
```suggestion
        redis_cmd_append_sstr_zval(&cmdstr, &args[i], redis_sock);
```

File: redis_commands.c [hunk 2]
```diff
@@ -3461,71 +3450,57 @@ int redis_hset_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
...
-        zend_string_release(zkey);
+        redis_cmd_append_sstr_key_zstr(&cmdstr, key, redis_sock, slot);
 
-        for (i = 1; i < argc; ++i) {
+        for (i = 0; i < argc; ++i) {
             if (i % 2) {
-                zkey = zval_get_string(&z_args[i]);
-                redis_cmd_append_sstr(&cmdstr, ZSTR_VAL(zkey), ZSTR_LEN(zkey));
-                zend_string_release(zkey);
+                redis_cmd_append_sstr_zval(&cmdstr, args, redis_sock);
```
Comment (yatsukhnenko) [inline redis_commands.c]:
```suggestion
                redis_cmd_append_sstr_zval(&cmdstr, &args[i], redis_sock);
```

File: redis_commands.c [hunk 3]
```diff
@@ -3461,71 +3450,57 @@ int redis_hset_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
...
             } else {
-                redis_cmd_append_sstr_zval(&cmdstr, &z_args[i], redis_sock);
+                redis_cmd_append_sstr_zval(&cmdstr, args, NULL);
```
Comment (yatsukhnenko) [inline redis_commands.c]:
```suggestion
                redis_cmd_append_sstr_zval(&cmdstr, &args[i], NULL);
```

File: redis_commands.c [hunk 4]
```diff
@@ -3461,71 +3450,57 @@ int redis_hset_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
...
             }
+            args++;
```
Comment (yatsukhnenko) [inline redis_commands.c]:
```suggestion
```

File: redis_commands.c [hunk 5]
```diff
@@ -3887,57 +3862,33 @@ int redis_sort_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
...
+    redis_cmd_append_sstr_key_zstr(&cmdstr, key, redis_sock, slot);
 
     // Iterate through the members we're removing
-    for (i = 1; i < argc; i++) {
-        zstr = zval_get_string(&z_args[i]);
-        redis_cmd_append_sstr(&cmdstr, ZSTR_VAL(zstr), ZSTR_LEN(zstr));
-        zend_string_release(zstr);
+    for (i = 0; i < argc; i++) {
+        redis_cmd_append_sstr_zval(&cmdstr, args, NULL);
+        args++;
```
Comment (yatsukhnenko) [inline redis_commands.c]:
```suggestion
        redis_cmd_append_sstr_zval(&cmdstr, &args[i], NULL);
```

**Review by yatsukhnenko — APPROVED** [review]
Suggested changes are not necessary, just my code-style preferences

**Comment by JakubOnderka** [comment]
Fixed, thanks for your hint, it looks better now.

**Review by michael-grunder — APPROVED** [review]

**Comment by michael-grunder** [comment]
I was unaware of these `_tmp_` variants and the codegen is better (at least with GCC on x86-64).

**Pull request merged**